test/parallel/test-stream-pipe-multiple-pipes.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+const common = require('../common');
+const stream = require('stream');
+const assert = require('assert');
+
+const readable = new stream.Readable({
+  read: () => {}
+});
+
+const writables = [];
+
+for (let i = 0; i < 5; i++) {
+  const target = new stream.Writable({
+    write: (chunk, encoding, callback) => {
+      target.output.push(chunk);
+      callback();
+    }
+  });
+  target.output = [];
+
+  target.on('pipe', common.mustCall(() => {}));
+  readable.pipe(target);
+
+
+  writables.push(target);
+}
+
+const input = Buffer.from([1, 2, 3, 4, 5]);
+
+readable.push(input);
+
+process.nextTick(common.mustCall(() => {
```

Comment:
can you also add a little comment why `process.nextTick` is needed here? For the sake of newbies.